modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -1704,29 +1704,63 @@ void simplifySubgraphs(opencv_onnx::GraphProto& net)
     simplifySubgraphs(Ptr<ImportGraphWrapper>(new ONNXGraphWrapper(net)), subgraphs);
 }
 
-Mat getMatFromTensor(const opencv_onnx::TensorProto& tensor_proto, bool uint8ToInt8)
+
+
+char* get_raw_data(const opencv_onnx::TensorProto& tensor_proto, const std::string base_path = ""){
+    if(tensor_proto.has_data_location() && tensor_proto.data_location() == opencv_onnx::TensorProto::EXTERNAL){
+        auto it_begin = tensor_proto.external_data().begin();
+        auto it_end = tensor_proto.external_data().end();
+        // file path
+        auto it = std::find_if(it_begin, it_end,[](const auto& entry) { return entry.key() == "location"; });
+        CV_Assert(it != it_end);
+
+        std::string location_path = it->value();
+        std::string full_path = base_path.empty() ? location_path : (base_path + "/" + location_path);
+
+        std::ifstream file(full_path, std::ios::binary | std::ios::ate);
+        CV_Assert(file.is_open());
+
+        std::streamsize size = file.tellg();
+        file.seekg(0, std::ios::beg);
+
+        char* buffer = new char[size];
+        if (!file.read(buffer, size)) {
+            delete[] buffer;
+            CV_Error(Error::StsError, "Failed to read external tensor data from file: " + std::string(it->value()));
```

Comment:
There are platforms, where filesystem is not available. You need to handle it like here: https://github.com/opencv/opencv/blob/d1b4b46dc61e3552b03bd440c39283a75d89f0a2/modules/core/src/utils/filesystem.cpp#L102.